### PR TITLE
Add `mode` argument to `visualize` for more compact data or task graph display

### LINKED ISSUE #82: Increase graph readability when using parametrized providers
Example:
image

We have no clear idea about a solution right now.

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Here is an before and after for ESSsans' `IofQ[SampleRun]`.
Before:
sciline-vis-before
After:
sciline-vis-after

Fixes #82.

### CODE REVIEW

File: src/sciline/visualize.py [hunk 1]
```diff
@@ -82,28 +113,72 @@ def _to_subgraphs(graph: FormattedGraph) -> dict[str, FormattedGraph]:
...
+                                arg.name,
+                                ret_name,
+                                lhead=subgraph.name,
+                                # Thick pen to indicate multiple connections
+                                penwidth='2.0',
+                            )
+                            cluster_connected.append(arg.name)
+                    else:
+                        dot.edge(arg.name, ret_name)
         # else: Do not draw dummy providers created by Pipeline when setting instances
```
Comment (nvaytet) [inline src/sciline/visualize.py:184]: There is an error in the built docs (in parameter tables):
Screenshot at 2024-10-30 09-24-00

File: src/sciline/visualize.py [hunk 2]
```diff
@@ -82,28 +113,72 @@ def _to_subgraphs(graph: FormattedGraph) -> dict[str, FormattedGraph]:
...
```
Comment (nvaytet) [inline src/sciline/visualize.py:183]: I don't think the curved arrows are always better. Comparing the new to old figure, while there are less boxes and duplication in the new figure, I find it harder to follow the arrows.

New:
Screenshot at 2024-10-30 09-25-48

Old:
Screenshot at 2024-10-30 09-26-06

File: src/sciline/visualize.py [hunk 3]
```diff
@@ -82,28 +113,72 @@ def _to_subgraphs(graph: FormattedGraph) -> dict[str, FormattedGraph]:
...
```
Comment (nvaytet) [inline src/sciline/visualize.py:182]: Another error in the generic providers notebook
Screenshot at 2024-10-30 09-29-38

File: src/sciline/visualize.py (hunk 2 above)
Comment (SimonHeybrock) [reply src/sciline/visualize.py:183]: Hmm, why are they curved now?

Comment (SimonHeybrock) [reply src/sciline/visualize.py:186]: Avoided some, see update.

**Review by nvaytet — APPROVED** [review]

**Pull request merged**